lib/peer-discovery.js

```diff
@@ -138,4 +157,59 @@ module.exports = class PeerDiscovery {
   }
 }
 
+class PeerDiscoverySession {
+  constructor (discovery, { server = true, client = true, onerror = safetyCatch }) {
+    this.discovery = discovery
+    this.isClient = client
+    this.isServer = server
+    this.destroyed = false
+
+    this._onerror = onerror
+
+    if (client) this.discovery._clientSessions++
+    if (server) this.discovery._serverSessions++
+
+    this.refresh().catch(this._onerror)
+  }
+
+  get swarm () {
+    return this.discovery.swarm
+  }
+
+  get topic () {
+    return this.discovery.topic
+  }
+
+  async refresh ({ client = this.isClient, server = this.isServer } = {}) {
+    if (this.destroyed) throw new Error('PeerDiscovery is destroyed')
+    if (!client && !server) throw new Error('Cannot refresh with neither client nor server option')
+
+    if (client !== this.isClient) {
+      this.discovery._clientSessions += client ? 1 : -1
+    }
+
+    if (server !== this.isServer) {
+      this.discovery._serverSessions += server ? 1 : -1
+    }
```

Comment:
what is the user call refresh manually on the session? wont that upset the count when decremented in destroy?

Reply:
Terribly good point, forgot to update `isClient` and `isServer`!

Reply:
e8c4fc15931af9b20a94a890fd2a4ed7948367df should fix that.